### PR TITLE
diy-data finished

### PR DESCRIPTION
**Pull request by mmarla**

### CODE REVIEW

File: src/kata2.booleanToWord.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const booleanToWord = (boolean) => {
-
+    return (boolean ? 'Yes' : 'No');
```
Comment (Daniel-Cross) [inline src/kata2.booleanToWord.js:2]: Really good use of a ternary operator. Concise and effective.

File: src/kata1.fizzBuzz.js [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
 
-}
+    if (number % 5 === 0) {
+        return 'Buzz'
+    }
+
+    if (number % 3 === 0) {
+        return 'Fizz'
+    }
+
```
Comment (Daniel-Cross) [inline src/kata1.fizzBuzz.js:13]: Instead of 3 separate if statements, one else if or switch statement would have been more concise here. Code works though and passes the tests.

File: src/kata3.numberToReversedDigits.js [hunk 3]
```diff
@@ -1,5 +1,4 @@
-const numberToReversedDigits = (number) => {
-
-}
+const numberToReversedDigits = (number) =>
+    number.toString().split('').map(Number).reverse();
```
Comment (Daniel-Cross) [inline src/kata3.numberToReversedDigits.js:2]: Very good, concise and effective.

File: src/kata4.humanCatDogYears.js [hunk 4]
```diff
@@ -1,5 +1,12 @@
...
+const humanCatDogYears = (number) => { 
+  let newArray = [number]
+  if (number === 1) {
+  return [1, 15, 15]
+  } if (number === 2) {
+  return [2, 24, 24]
+  } newArray.push(24 + ((number - 2) * 4))
+  newArray.push(24 + ((number - 2) * 5))
+  return newArray
 }
```
Comment (Daniel-Cross) [inline src/kata4.humanCatDogYears.js:10]: Although this code passes the tests you wrote, unfortunately it doesn't stand up to any robust or dynamic testing. If the user enters a number you haven't hardcoded in here, the code will fail as will the tests.

File: __tests__/kata4.humanCatDogYears.test.js [hunk 5]
```diff
@@ -1,3 +1,11 @@
...
-// Look Ma, no handlebars!!!
+describe('humanCatDogYears', () => {
+    it('returns array of cat, human and dog ages', () => {
+        expect(humanCatDogYears(10)).toEqual([10, 56, 64])
+    });
+
+    it('returns array of cat, human and dog ages even if less than 3', () => {
+        expect(humanCatDogYears(2)).toEqual([2, 24, 24])
+    });
+});
```
Comment (Daniel-Cross) [inline __tests__/kata4.humanCatDogYears.test.js:11]: These tests are fine and work accordingly but should the user enter a 0, negative number, a 3 or upwards, the code that passes these tests will fail. Ensure your tests have more robust coverage.

File: src/kata5.reachDestination.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
-const reachDestination = (distance, speed) => {
-
+const reachDestination = (distance, speed) => { 
+	return `I should be there in ${Math.round((distance / speed) * 2) / 2} hours`;
 }
```
Comment (Daniel-Cross) [inline src/kata5.reachDestination.js:3]: Code passes the test that was written but could be more accurate.

File: __tests__/kata5.reachDestination.test.js [hunk 7]
```diff
@@ -2,6 +2,6 @@ const { reachDestination } = require('../src');
 
 describe('reachDestination', () => {
   it('returns string with estimated time of arrival', () => {
-    
+    expect(reachDestination(44, 10)).toEqual('I should be there in 4.5 hours')
   });
 });
```
Comment (Daniel-Cross) [inline __tests__/kata5.reachDestination.test.js:7]: More robust testing needed for edge cases.

File: src/kata6.joinNames.js [hunk 8]
```diff
@@ -1,5 +1,8 @@
 const joinNames = (namesObj) => {
-
+    let names = namesObj.map(a => a.name);
+    if (names.length === 1) return namesObj[0].name;
+    let lastName = names.pop();
+    return names.join(', ') + ' & ' + lastName;
 }
 
-module.exports = joinNames;
+module.exports = joinNames
```
Comment (Daniel-Cross) [inline src/kata6.joinNames.js:8]: Works well. Effective and clean code.

File: src/kata7.getEmployerRole.js [hunk 9]
```diff
@@ -1,5 +1,9 @@
 const getEmployerRole = (employeeName, employees) => {
-
+    for (var i = 0; i < employees.length; i++) {
+        if (employees[i].name === employeeName) {
+            return employees[i].role;
+        }
+    }
 }
```
Comment (Daniel-Cross) [inline src/kata7.getEmployerRole.js:7]: Really concise and effective code. Well done.
